### PR TITLE
MAPI-1022 Upgrade the common notifications api client

### PR DESCRIPTION
**Pull request by DiegoVazquezNanini**

### CODE REVIEW

**Comment by davidfurey** [comment]
:+1: 
Looks fine to me, but might be worth @alexduf taking a quick look too

**Comment by alexduf** [comment]
It looks good to me :+1: , but I think it's missing the two steps generation of the payload.
The goal of that first change is to create a generic payload (in your case com.gu.mobile.notifications.client.models.ContentAlertPayload) then generate the Android and iOS payload from that object.

It's not a blocker and could be done either as a new ticket, or I'll integrate it into MAPI-1083

**Comment by DiegoVazquezNanini** [comment]
Thanks @alexduf. 
No problem, I can integrate the work for MAPI-1083 into this PR.

File: src/main/scala/com/gu/mobile/notifications/content/lib/Futures.scala [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package com.gu.mobile.notifications.content.lib
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
+import com.gu.management.TimingMetric
+
+object Futures {
+
+  implicit class RichFuture[A](val future: Future[A]) {
+    def measure(f: (Try[A], Long) => Unit)(implicit context: ExecutionContext) {
```
Comment (agulowaty) [inline src/main/scala/com/gu/mobile/notifications/content/lib/Futures.scala]: private ?

**Comment by DiegoVazquezNanini** [comment]
@alexduf I think I should do the work to produce a generic payload on a different story.

**Comment by alexduf** [comment]
No worries!

**Pull request closed without merging**